### PR TITLE
[enh] use generic function for symlink creation.

### PR DESCRIPTION
**Pull request by M5oul**
Simplify the code using a generic function for symlink creation.
I tested for SSOwat, yunohost and moulinette packages and it works perfectly.

### CODE REVIEW

**Comment by M5oul** [comment]
Up!

File: ynh-dev [hunk 1]
```diff
@@ -182,87 +191,52 @@ elif [ "$1" = "use-git" ]; then
...
                 ;;
             yunohost)
                 echo "Using Git repository for yunohost"
 
                 # bin
-                if [ ! -L '/usr/bin/yunohost' ]; then sudo rm /usr/bin/yunohost; fi
-                sudo ln -sfn /vagrant/yunohost/bin/yunohost /usr/bin/yunohost
-                if [ ! -L '/usr/bin/yunohost-api' ]; then sudo rm /usr/bin/yunohost-api; fi
-                sudo ln -sfn /vagrant/yunohost/bin/yunohost-api /usr/bin/yunohost-api
+		create_sym_link "/usr/bin/yunohost" "/vagrant/yunohost/bin/yunohost"
```
Comment (opi) [inline ynh-dev]: incorrect order

File: ynh-dev [hunk 2]
```diff
@@ -182,87 +191,52 @@ elif [ "$1" = "use-git" ]; then
...
+		create_sym_link "/usr/bin/yunohost-api" "/vagrant/yunohost/bin/yunohost-api"
```
Comment (opi) [inline ynh-dev]: incorrect order

File: ynh-dev [hunk 3]
```diff
@@ -182,87 +191,52 @@ elif [ "$1" = "use-git" ]; then
...
-                sudo ln -sfn /vagrant/yunohost/data/hooks /usr/share/yunohost/hooks
-                if [ ! -L '/usr/share/yunohost/templates' ]; then sudo rm -rf /usr/share/yunohost/templates; fi
-                sudo ln -sfn /vagrant/yunohost/data/templates /usr/share/yunohost/templates
-                if [ ! -L '/usr/share/yunohost/helpers' ]; then sudo rm /usr/share/yunohost/helpers; fi
-                sudo ln -sfn /vagrant/yunohost/data/helpers /usr/share/yunohost/helpers
-                if [ ! -L '/usr/share/yunohost/helpers.d' ]; then sudo rm -rf /usr/share/yunohost/helpers.d; fi
-                sudo ln -sfn /vagrant/yunohost/data/helpers.d /usr/share/yunohost/helpers.d
-                if [ ! -L '/usr/share/yunohost/yunohost-config/moulinette' ]; then sudo rm -rf /usr/share/yunohost/yunohost-config/moulinette; fi
-                sudo ln -sfn /vagrant/yunohost/data/other /usr/share/yunohost/yunohost-config/moulinette
+                create_sym_link "/etc/bash_completion.d/yunohost" "/vagrant/yunohost/data/bash-completion.d/yunohost"
```
Comment (opi) [inline ynh-dev]: incorrect order

File: ynh-dev [hunk 4]
```diff
@@ -182,87 +191,52 @@ elif [ "$1" = "use-git" ]; then
...
+		create_sym_link "/usr/share/moulinette/actionsmap/yunohost.yml" "/vagrant/yunohost/data/actionsmap/yunohost.yml"
```
Comment (opi) [inline ynh-dev]: incorrect order

File: ynh-dev [hunk 5]
```diff
@@ -182,87 +191,52 @@ elif [ "$1" = "use-git" ]; then
...
+		create_sym_link "/usr/share/yunohost/hooks" "/vagrant/yunohost/data/hooks"
```
Comment (opi) [inline ynh-dev]: incorrect order

File: ynh-dev [hunk 6]
```diff
@@ -182,87 +191,52 @@ elif [ "$1" = "use-git" ]; then
...
+		create_sym_link "/usr/share/yunohost/templates" "/vagrant/yunohost/data/templates"
```
Comment (opi) [inline ynh-dev]: incorrect order

File: ynh-dev [hunk 7]
```diff
@@ -182,87 +191,52 @@ elif [ "$1" = "use-git" ]; then
...
+		create_sym_link "/usr/share/yunohost/helpers" "/vagrant/yunohost/data/helpers"
```
Comment (opi) [inline ynh-dev]: incorrect order

File: ynh-dev [hunk 8]
```diff
@@ -182,87 +191,52 @@ elif [ "$1" = "use-git" ]; then
...
+		create_sym_link "/usr/share/yunohost/helpers.d" "/vagrant/yunohost/data/helpers.d"
```
Comment (opi) [inline ynh-dev]: incorrect order

File: ynh-dev [hunk 9]
```diff
@@ -182,87 +191,52 @@ elif [ "$1" = "use-git" ]; then
...
+		create_sym_link "/usr/share/yunohost/yunohost-config/moulinette" "/vagrant/yunohost/data/other"
```
Comment (opi) [inline ynh-dev]: incorrect order

File: ynh-dev [hunk 10]
```diff
@@ -182,87 +191,52 @@ elif [ "$1" = "use-git" ]; then
...
 
                 # debian
-                if [ ! -L '/usr/share/pam-configs/mkhomedir' ]; then sudo rm /usr/share/pam-configs/mkhomedir; fi
-                sudo ln -sfn /vagrant/yunohost/debian/conf/pam/mkhomedir /usr/share/pam-configs/mkhomedir
+		create_sym_link "/usr/share/pam-configs/mkhomedir" "/vagrant/yunohost/debian/conf/pam/mkhomedir"
```
Comment (opi) [inline ynh-dev]: incorrect order

File: ynh-dev [hunk 11]
```diff
@@ -182,87 +191,52 @@ elif [ "$1" = "use-git" ]; then
...
-                sudo ln -sfn /vagrant/yunohost/lib/metronome/modules/mod_auth_ldap2.lua /usr/lib/metronome/modules/mod_auth_ldap2.lua
-                if [ ! -L '/usr/lib/metronome/modules/mod_legacyauth.lua' ]; then sudo rm /usr/lib/metronome/modules/mod_legacyauth.lua; fi
-                sudo ln -sfn /vagrant/yunohost/lib/metronome/modules/mod_legacyauth.lua /usr/lib/metronome/modules/mod_legacyauth.lua
-                if [ ! -L '/usr/lib/metronome/modules/mod_storage_ldap.lua' ]; then sudo rm /usr/lib/metronome/modules/mod_storage_ldap.lua; fi
-                sudo ln -sfn /vagrant/yunohost/lib/metronome/modules/mod_storage_ldap.lua /usr/lib/metronome/modules/mod_storage_ldap.lua
-                if [ ! -L '/usr/lib/metronome/modules/vcard.lib.lua' ]; then sudo rm /usr/lib/metronome/modules/vcard.lib.lua; fi
-                sudo ln -sfn /vagrant/yunohost/lib/metronome/modules/vcard.lib.lua /usr/lib/metronome/modules/vcard.lib.lua
+		create_sym_link "/usr/lib/metronome/modules/ldap.lib.lua" "/vagrant/yunohost/lib/metronome/modules/ldap.lib.lua"
```
Comment (opi) [inline ynh-dev]: incorrect order, + nexts

**Comment by M5oul** [comment]
Ok, I fixed the order and use spaces instead of tabulations.

**Comment by M5oul** [comment]
I tested for SSOwat, yunohost and moulinette packages and it works fine.

**Pull request merged**